### PR TITLE
Add support for Date objects

### PR DESCRIPTION
**Pull request by ylt**
- Generate Date type
- Sort the types in priority order (DateTimes need parsing before we get onto parsing Dates)

### CODE REVIEW

File: src/generators/php/index.js [hunk 1]
```diff
@@ -304,6 +305,35 @@ class PHP extends Generator {
     return types.length > 1 ? `${types.join("|")}` : types[0];
   }
 
+  validationTypeWeight(prefixedTypeName, isExtension, model) {
+    let typeName = this.getPropNameFromFQP(prefixedTypeName);
+    let compactedTypeName = this.getCompacted(prefixedTypeName);
+    let extension = this.extensions[model.extensionPrefix];
+    switch (typeName) {
+      case "Boolean":
+        return "bool";
```
Comment (thtg88) [inline src/generators/php/index.js]: Should this be `0`?

Comment (ylt) [reply src/generators/php/index.js]: Lol, oops - yeah probably

File: src/generators/php/index.js [hunk 2]
```diff
@@ -304,6 +305,35 @@ class PHP extends Generator {
...
+        return 6;
+      case "Property":
+      case "Text":
+      case "URL":
+        return 5;
+      case "Duration":
+        return 4;
+      case "null":
+        return 8;
+    }
```
Comment (thtg88) [inline src/generators/php/index.js]: Is it worth putting a default weight of 9 for classes that are not of native types? Or am I misunderstanding 🤔

Comment (ylt) [reply src/generators/php/index.js]: I think they'll automatically get the equivalent of 0

Comment (ylt) [reply src/generators/php/index.js]: openactiv… 2020-03-12 17-14-14

**Comment by nickevansuk** [comment]
Just a note to pull the latest model files and beta before publishing this to packagist - as they’ve been updated again

**Pull request merged**